example/gpu_memory_profiler/profile_analyzer.py

```diff
@@ -0,0 +1,103 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""profile_analyzer.py
+
+   Analyze the memory profile in 'SETME.py'.
+"""
+import csv
+import logging
+
+
+def analyze(memory_profile_name,
+            layer_kw_dict,
+            data_struct_kw_dict,
+            expected_sum=None,
+            warn_kw_404=False):
+    """
+    Analyze the memory profile.
+    """
+    layer_wise_memory_profile, data_struct_wise_memory_profile = {}, {}
+
+    with open(memory_profile_name, 'r') as csvfile:
+        reader = csv.reader(csvfile)
+
+        trackable_memory_alloc = 0.0
+
+        tag, memory_alloc = "unknown", 0.0
+
+        for row in reader:
+            tag, memory_alloc = row[0], float(row[1]) / 1024
+
+            def _internal_categorize(kw_dict, memory_profile):
+
+                def _truncate(tag):
+                    return tag[:50] + '..' if len(tag) > 50 else tag
+
+                kw_matched = None
+
+                for key, val in kw_dict.items():
+                    for kw in val:
+                        if kw in tag:
+                            if kw_matched is not None:
+                                # pylint: disable=logging-not-lazy
+                                logging.warning("Tag %s is considered match for "
```

Comment:
you can directly pass the strings as additional arguments to the logging function, no need to format since using %s already.